src/components/deformation/block-inputs/block-inputs-menu.tsx

```diff
@@ -64,10 +64,15 @@ export default class BlockInputsMenu extends BaseComponent<IProps, IState> {
           top={0}
           color={blockInputButtonColor}
           onClick={this.toggleBlockInputs}
-          disabled={running ? true : false}
+          disabled={running || !deformationHistory.length ? true : false}
           running={running}
         >
-          <ButtonText running={running}>Block Inputs</ButtonText>
+          <ButtonText
+            running={running}
+            disabled={running || !deformationHistory.length ? true : false}
```

Comment:
I think these logic expressions are unnecessarily complicated.   These are the same I think:

```javascript

> x = false || !0 ? true : false
true
> x = false || !0
true
> x = false || !1 ? true : false
false
> x = false || !1
false

```